File: library/src/main/java/cn/wandersnail/widget/listview/InflatedViewHolder.java
package cn.wandersnail.widget.listview;

import android.content.Context;
import android.view.View;

import androidx.annotation.IdRes;
import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;

/**
 * 根据布局ID自动填充
 * <p>
 * date: 2019/8/22 19:08
 * author: zengfansheng
 */
public abstract class InflatedViewHolder<T> implements BaseViewHolder<T> {
    private final View rootView;

    public InflatedViewHolder(@NonNull Context context, @LayoutRes int resId) {
        rootView = View.inflate(context, resId, null);
    }

    @NonNull
    @Override
    public View createView() {
        return rootView;
    }

    public <V extends View> V getView(@IdRes int resId) {
        return rootView.findViewById(resId);
    }
}

File: library/src/main/java/com/snail/widget/listview/InflatedViewHolder.kt
package com.snail.widget.listview

import android.content.Context
import android.view.View
import androidx.annotation.IdRes
import androidx.annotation.LayoutRes

/**
 * 自动填充
 *
 * date: 2019/5/1 15:52
 * author: zengfansheng
 */
abstract class InflatedViewHolder<T>(context: Context, @LayoutRes private val layoutResId: Int) : BaseViewHolder<T>() {
    val rootView: View = View.inflate(context, layoutResId, null)

    override fun createView(): View {
        return rootView
    }
    
    fun <V : View> getView(@IdRes resId: Int): V {
        return rootView.findViewById(resId)
    }
}